waspc/data/Generator/templates/server/src/app.js

```diff
@@ -35,7 +35,8 @@ app.use((err, req, res, next) => {
     return res.status(err.statusCode).json({ message: err.message, data: err.data })
   }
 
-  return next(err)
+  console.error(err);
+  return res.status(500).json({message: "An internal server error occurred."})
```

Comment:
What really confuses me is that I was 100% sure we already had this in place!

Look, even the docs say it: https://wasp-lang.dev/docs/language/features#error-handling .

I looked at older code but we never had explicit handling of this.

So I checked express docs: https://expressjs.com/en/guide/error-handling.html#the-default-error-handler .

So from what I get, by default they will:
1. Set status message to 500 if message doesn't have status that is in 400-599 range.
2. Set error message to an HTML description of its status if in production, or stack if in dev. So maybe that is why you saw stack, while in production it wouldn't be visible

Based on that, I think it best that we don't touch anything!

But, I do agree this is completely unclear from the code. Maybe we should, above `return next(err)` line, put a link to this piece in docs and quickly comment what it does?

Here is except from their docs:

> If you pass an error to next() and you do not handle it in a custom error handler, it will be handled by the built-in error handler; the error will be written to the client with the stack trace. The stack trace is not included in the production environment.
>
> Set the environment variable NODE_ENV to production, to run the app in production mode.
>
> When an error is written, the following information is added to the response:
> 
> The res.statusCode is set from err.status (or err.statusCode). If this value is outside the 4xx or 5xx range, it will be set to 500.
The res.statusMessage is set according to the status code.
The body will be the HTML of the status code message when in production environment, otherwise will be err.stack.
Any headers specified in an err.headers object.
If you call next() with an error after you have started writing the response (for example, if you encounter an error while streaming the response to the client) the Express default error handler closes the connection and fails the request.

Reply:
Ah, duh! Ok, thanks. That is embarrassing. I didn't even stop to think it was env-based, which obviously makes sense. The Issue and our chat had me thinking it had to be on us somehow lol but ok yeah, in prod it would work. I will update it with the link to the default error handler, thanks! 👍🏻

Reply:
The minor downside is it returns HTML and not JSON, but that is fine for now I think.